src/command/logical.c

```diff
@@ -0,0 +1,68 @@
+/*
+** EPITECH PROJECT, 2022
+** 42sh
+** File description:
+** Logical function checkers
+*/
+
+#include <stddef.h>
+#include <stdio.h>
+#include <string.h>
+#include "logical.h"
+#include "messages.h"
+
+static bool check_logical(command_t *command, bool empty)
+{
+    command_t *prev = command->prev;
+    separatour_next_type_t next = command->separator_next;
+
+    if (empty && next == OR) {
+        fprintf(stderr, "%s\n", MISSING_COMMAND);
+        return (false);
+    }
+    if (empty && prev != NULL) {
+        next = prev->separator_next;
+        if (next == OR || next == AND) {
+            fprintf(stderr, "%s\n", MISSING_COMMAND);
+            return (false);
+        }
+    }
+    return (true);
+}
+
+bool check_logicals(command_t *list, shell_t *shell)
+{
+    bool empty = false;
+    command_t *current = list;
+
+    while (current != NULL) {
```

Comment:
could use a for but ok

Reply:
> could use a for but ok

Nope for is overkill in this case, it would be ugly if we did:
`for (command_t *current = list; current != NULL; current = current->next)`
(ew)

Reply:
> Nope for is overkill in this case, it would be ugly if we did: `for (command_t *current = list; current != NULL; current = current->next)` (ew)

Why overkill ? Here we need to init a variable, loop through the list and make a specific action every time.... exactly what a for does, but anyway it works as I said

Reply:
> > Nope for is overkill in this case, it would be ugly if we did: `for (command_t *current = list; current != NULL; current = current->next)` (ew)
> 
> Why overkill ? Here we need to init a variable, loop through the list and make a specific action every time.... exactly what a for does, but anyway it works as I said

Mb I didn't mean overkill I wanted to say that it would be ugly xD